include/nbl/video/ILogicalDevice.h

```diff
@@ -0,0 +1,430 @@
+#ifndef __NBL_I_GPU_LOGICAL_DEVICE_H_INCLUDED__
+#define __NBL_I_GPU_LOGICAL_DEVICE_H_INCLUDED__
+
+#include "nbl/core/IReferenceCounted.h"
+#include "nbl/video/IGPUQueue.h"
+#include "nbl/video/IGPUSemaphore.h"
+#include "nbl/video/IDescriptorPool.h"
+#include "nbl/video/IGPUDescriptorSet.h"
+#include "nbl/video/IGPUCommandPool.h"
+#include "nbl/video/IGPUFramebuffer.h"
+#include "nbl/video/IGPUGraphicsPipeline.h"
+#include "nbl/video/ISwapchain.h"
+
+namespace nbl {
+namespace video
+{
+
+class ILogicalDevice : public core::IReferenceCounted
+{
+public:
+    struct SQueueCreationParams
+    {
+        IGPUQueue::E_CREATE_FLAGS flags;
+        uint32_t familyIndex;
+        uint32_t count;
+        const float* priorities;
+    };
+    struct SCreationParams
+    {
+        uint32_t queueParamsCount;
+        const SQueueCreationParams* queueCreateInfos;
+        // ???:
+        //uint32_t enabledExtensionCount;
+        //const char* const* ppEnabledExtensionNames;
+        //const VkPhysicalDeviceFeatures* pEnabledFeatures;
```

Comment:
we need our own feature enum (or two enums one for extensions one for enums) that the backends can map to OpenGL(ES) and Vulkan extensions or features

We're looking for things like:
- KHR Raytracing
- Multidraw Indirect
- Parameter buffers for Multidraw Indirect
- Geom, Tessellation shaders
- one enum per each VK GLSL thing we report with our compiler

The `IAPIConnection` needs to  report the supported list so we know what we can use.

there's a very incomplete list in `queryFeature` of the old `COpenGLDriver`

Reply:
@achalpandeyy feature and extension querying